src/espcomp/mod.rs

```diff
@@ -0,0 +1,129 @@
+use std::path::PathBuf;
```

Comment:
If we are expecting the `espcomp` module to be a single file, please rather than using a directory named `espcomp` and then a lonely `mod.rs` file inside, just switch to a single `espcomp.rs` file. Should you need inner - and long - sub-modules that need to live in their own files, you can still create an `espcomp` subdirectory next to the `espcomp` file and place these inside. But I feel that the `mod.rs` is not practiced that much these days. Maybe subjective.

Reply:
I actually forgot to add the serde models to git here, so there is a second file (will push shortly). I also have two modules for checksum hashing, lock-files, etc. that I omitted from the PR in order to keep the concept simple for now.